astropy/table/table.py

```diff
@@ -3110,11 +3110,16 @@ def rename_columns(self, names, new_names):
 
     def _set_row(self, idx, colnames, vals):
         try:
-            assert len(vals) == len(colnames)
```

Comment:
Instead of this entire change just do:
```
if len(vals) != len(colunames):
    raise Exception
```
This keeps the original logic and exception messages. In this case exception chaining is NOT desired.

Reply:
@mhvk  ?

Reply:
@mhvk - am I missing something? I think that my suggestion *exactly replicates* the original. In the original, if either `vals` or `columns` has no length then you get a `TypeError` and this re-raises the `ValueError` below which you get for differing lengths.

My version is basically just replacing `AssertionError` with `Exception`. In fact we could make it precisely the same as the original by using `raise AssertionError` which might be a good idea to convey the intent here of an assertion.

Reply:
Ah, I failed to take your suggestion literally, as it was intended. Yes, might as well keep the logic identical. And then indeed a `from None` to avoid the chaining.